tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -1532,4 +1532,88 @@ public function theResponseShouldContainTheFollowingDriveInformation(TableNode $
 			throw new Error('Response is not an array or the array does not consist key "drive"');
 		}
 	}
+
+	/**
+	 * @When user :user gets details of a group :groupName using the Graph API
+	 *
+	 * @param string $user
+	 * @param string $groupName
+	 *
+	 * @return void
+	 */
+	public function userGetsDetailsOfAGroupUsingTheGraphApi(string $user, string $groupName): void {
+		$credentials = $this->getAdminOrUserCredentials($user);
+		try {
+			$groupId = $this->getGroupIdByName($groupName);
+		} catch (Exception $e) {
+			$groupId = WebDavHelper::generateUUIDv4();
+		}
+
+		$this->featureContext->setResponse(
+			GraphHelper::getSingleGroup(
+				$this->featureContext->getBaseUrl(),
+				$groupId,
+				$this->featureContext->getStepLineRef(),
+				$credentials["username"],
+				$credentials["password"]
+			)
+		);
+	}
+
+	/**
+	 * Get Group Id from response
+	 *
+	 * @param string $groupName - name of the group
+	 * @param array|null $allGroups
+	 *
+	 * @return string - id of the group
+	 * @throws Exception | GuzzleException
+	 */
+	public function getGroupIdByName(string $groupName, ?array $allGroups = null): string {
+		if ($allGroups === null) {
+			$allGroups = $this->getArrayOfGroupsResponded($this->listGroups());
+		}
+		foreach ($allGroups as $group) {
+			if ($group['displayName'] === $groupName) {
+				return $group['id'];
+			}
+		}
+		throw new Exception("Group with name '$groupName' not found in the groups list");
+	}
+
+	/**
+	 * @Then the json responded should contain a group with these key and value pairs:
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function theJsonRespondedShouldContainAGroupWithTheseKeyAndValuePairs(TableNode $table): void {
```

Comment:
What about using the new JSON response drill-down functionality? Would that make sense @SagarGi ?

Reply:
yes we can go for it as we have already implemented for `apiCapabilities` suite